diff --git a/ShellPkg/ShellPkg.dsc b/ShellPkg/ShellPkg.dsc
index dd0d88603f..be93c3de1d 100644
--- a/ShellPkg/ShellPkg.dsc
+++ b/ShellPkg/ShellPkg.dsc
@@ -65,6 +65,10 @@
   DxeServicesLib|MdePkg/Library/DxeServicesLib/DxeServicesLib.inf
   ReportStatusCodeLib|MdePkg/Library/BaseReportStatusCodeLibNull/BaseReportStatusCodeLibNull.inf
 
+  ####### ARM BSA ##############
+  BsaValLib|ShellPkg/Application/server-soc-ts/val/BsaValLib.inf
+  BsaPalLib|ShellPkg/Application/server-soc-ts/platform/pal_uefi_acpi/BsaPalLib.inf
+
 [LibraryClasses.ARM,LibraryClasses.AARCH64]
   #
   # It is not possible to prevent the ARM compiler for generic intrinsic functions.
@@ -88,6 +92,7 @@
   # Build all the libraries when building this package.
   # This helps developers test changes and how they affect the package.
   #
+
   ShellPkg/Library/UefiShellLib/UefiShellLib.inf
   ShellPkg/Library/UefiShellAcpiViewCommandLib/UefiShellAcpiViewCommandLib.inf
   ShellPkg/Library/UefiShellCommandLib/UefiShellCommandLib.inf
@@ -156,5 +161,8 @@
   ShellPkg/DynamicCommand/DpDynamicCommand/DpApp.inf
   ShellPkg/Application/AcpiViewApp/AcpiViewApp.inf
 
+  ######### BSA #######
+  ShellPkg/Application/server-soc-ts/uefi_app/BsaAcs.inf
+
 [BuildOptions]
   *_*_*_CC_FLAGS = -D DISABLE_NEW_DEPRECATED_INTERFACES
